### ISSUE TITLE
Release 0.7.1 does not work with iojs

### ISSUE THREAD

**Issue by dugword** [mcavage/node-ldapjs#245]
The most recent release on npm, 0.7.1, does not work with iojs. It throws an error about closeSocket being undefined on lib/client/client.js:111.

This error seems to be resolved in the master branch.

Do you have an ETA on the next release?

Thank you

**Comment by pfmooney** [comment]
There are a few more things I want to tackle before cutting 0.8.
Sadly my focus has been on other things since autumn.

**Comment by flyswatter** [comment]
I also would love iojs support here.  I'll fork the master branch for now, but otherwise this is one of my last dependencies I need before I can use ES6 on node with no problems!  That's exciting!  If you're not excited about ES6 yet, you should try this [interactive workshop](https://github.com/domenic/count-to-6) to tour some of the new features, it's a lot of fun.

**Comment by AlexCline** [comment]
I also ran into this error running node v0.12.0.

```
TypeError: Cannot read property 'on' of undefined
    at setupSocket (/Users/alexcline/Development/cnc/node_modules/ldapjs/lib/client/client.js:111:14)
    at Client._connect (/Users/alexcline/Development/cnc/node_modules/ldapjs/lib/client/client.js:742:3)
    at new Client (/Users/alexcline/Development/cnc/node_modules/ldapjs/lib/client/client.js:247:22)
    at Object.createClient (/Users/alexcline/Development/cnc/node_modules/ldapjs/lib/client/index.js:60:12)
```

**Comment by jonekdahl** [comment]
I also ran into this same issue when trying to upgrade to Node.js 0.12.0. Just let me know if you need more information, or if there is anything I can do to help get ldapjs 0.8 out the door. Thanks!

**Comment by feisley** [comment]
Just realized that the issue described here is the same as what I just reported in #256

**Comment by frebos88** [comment]
This problem is breaking both passport-windowsauth and node-activedirectory when forced to use secure ldap. Would love for the quick patch to be merged.

**Comment by flyswatter** [comment]
Holding these on the master branch but not pushing to npm is some of the strangest maintainer behavior I've seen.

Regardless, the workaround I'm using is fairly pleasant, and involves pointing my package.json directly at this master branch:
'''
    "ldapjs": "mcavage/node-ldapjs",
'''

**Comment by pfmooney** [comment]
Due to the level of churn in the codebase since 0.7.1, I've wanted to stabilize the breaking changes before making a new release.  I agree, the delay has been unfortunate, but other projects have been dominating my time lately.  That said, I would rather wait to ensure the changes have been completely vetted rather than hastily cutting a release only to discover needed fixes which require other API breaks.

**Comment by flyswatter** [comment]
I appreciate that you're looking out for overall stability, and it's a great module, I just think there has to be a way where you don't need to wait for large other changes before pushing small hot fixes.

Have you considered using a model where `master` is reserved for production-ready changes, and larger upgrades are kept on other branches until they're ready to merge?  It's been working well for me on my work projects, and it seems like it would solve this issue.

**Comment by pfmooney** [comment]
It's less about changes being production ready and more about being willing to commit to the API changes that presently sit in master.  Fortunately I have made time to review some of the changes recently and things are looking positive in that regard.  Most of what's left is doing proper documentation.

**Comment by tamsler** [comment]
Is there a status update on this?

**Comment by lklepner** [comment]
I'm interested as well.

**Comment by pentode** [comment]
I've implemented a fix for this that works with both nodejs and iojs:

```
  // Since tls.socket doesn't emit 'close' events on older versions of node/io, we must register to receive
  // them on net.socket instead
  var closeSocket = ((opts.secure && EventEmitter.listenerCount(socket,'close')===0) ? socket.socket : socket);

```

**Comment by pfmooney** [comment]
I'm narrowing down the list for the 0.8 release.  See #270.

**Comment by jonekdahl** [comment]
@pfmooney Sorry, I'm confused, will this issue be fixed in 0.8?

**Comment by butterkekstorte** [comment]
I created a fork and fixed this issue, by replacing one line.
If you want to use it just require ldapjs-hotfix instead of ldapjs in your code and package.json with a version of 0.7.3.
